File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    if(pricesSize==0)return 0;

    int min=prices[0];
    int maxprofit=0;

    for(int i=0;i<pricesSize;i++){
        if(prices[i]<min){
            min=prices[i];
        }
        int profit=prices[i]-min;
        if(profit>maxprofit){
            maxprofit=profit;
        }
    }
    return maxprofit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min=prices[0];
        int maxprofit=0;
        for(int price:prices){
            min=Math.min(price,min);
            int profit=price-min;
            maxprofit=Math.max(profit,maxprofit);
        }
        return maxprofit;
    }
}